lib/std/math/powi.zig

```diff
@@ -10,108 +10,92 @@ const testing = std.testing;
 
 /// Returns the power of x raised by the integer y (x^y).
 ///
+/// Errors:
+///  - Overflow: Integer overflow or Infinity
+///  - Underflow: Absolute value of result smaller than 1
 /// Special Cases:
-///  - powi(x, +-0)   = 1 for any x
-///  - powi(0, y)     = 0 for any y
-///  - powi(1, y)     = 1 for any y
-///  - powi(-1, y)    = -1 for y an odd integer
-///  - powi(-1, y)    = 1 for y an even integer
-///  - powi(x, y)     = Overflow for y >= @sizeOf(x) - 1 or y > 0
-///  - powi(x, y)     = Underflow for y > @sizeOf(x) - 1 or y < 0
+///  - powi(x, 0)   = 1 unless T is i1, i0, u0
```

Comment:
nit: please use ascending sorting by x, then y.
T is not explained before `T is i1, i0, u0`. I guess its the type for x, but then I still dont understand what the result semantics are from the text.

Reply:
It is ordered by priority, added comment

Reply:
Are those missing u1?

Reply:
`u1` can represent `1`, so no need for special case